pilot/pkg/networking/core/v1alpha3/name_table.go

```diff
@@ -0,0 +1,109 @@
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha3
+
+import (
+	"istio.io/istio/pilot/pkg/model"
+	nds "istio.io/istio/pilot/pkg/proto"
+	"istio.io/istio/pilot/pkg/serviceregistry"
+	"istio.io/istio/pkg/config/constants"
+)
+
+// BuildNameTable produces a table of hostnames and their associated IPs that can then
+// be used by the agent to resolve DNS. This logic is always active. However, local DNS resolution
+// will only be effective if DNS capture is enabled in the proxy
+func (configgen *ConfigGeneratorImpl) BuildNameTable(node *model.Proxy, push *model.PushContext) *nds.NameTable {
+	if node.Type != model.SidecarProxy {
+		// DNS resolution is only for sidecars
+		return nil
+	}
+
+	out := &nds.NameTable{
+		Table: map[string]*nds.NameTable_NameInfo{},
+	}
+
+	for _, svc := range push.Services(node) {
```

Comment:
Is it possible to share code between existing dns_listener and this? And if it is - is it worth doing rather than having similar code at both places?

Reply:
So my goal was to nuke the existing dns listener as we are not going to use it. Should I do it as part of this PR?

Reply:
No. If our ultimate goal is to nuke existing DNS listener - just leave it as is and we can nuke it later.